sklearn/utils/_array_api.py

```diff
@@ -1101,3 +1101,21 @@ def _tolist(array, xp=None):
         return array.tolist()
     array_np = _convert_to_numpy(array, xp=xp)
     return [element.item() for element in array_np]
+
+
+def _nan_to_num(array, xp=None):
+    """Substitutes NaN values with 0 and inf values with the maximum or minimum
+    numbers available for the dtype respectively; like np.nan_to_num."""
+    if xp is None:
```

Comment:
We don't really need this. `get_namespace` handles the fact that we already have an xp defined so it returns that.
```suggestion
```